ddp-workspace/projects/ddp-sdk/src/lib/components/internationalization/languageSelector.component.ts

```diff
@@ -234,4 +237,21 @@ export class LanguageSelectorComponent implements OnInit, OnDestroy {
     });
     this.anchor.addNew(sub);
   }
+
+  private i18nListener(): void {
+    const sub = merge(
+      of(this.language.getCurrentLanguage()),
+      this.language.onLanguageChange().pipe(map(event => event.lang))
+    ).subscribe(languageCode => {
+      const direction = this.config.rtlLanguages.includes(languageCode) ? 'rtl' : 'ltr';
+      this.changeDocumentLanguage(direction, languageCode);
+    });
+    this.anchor.addNew(sub);
+  }
+
+  private changeDocumentLanguage(direction: string, languageCode: string): void {
+    const html = this.document.querySelector('html');
+    html.setAttribute('dir', direction);
+    html.setAttribute('lang', languageCode);
+  }
```

Comment:
Fixed it for me! I don't understand what was wrong with your original code; looked good to me. But this definitely works now.